internal/provider/dedicated_cluster_resource.go

```diff
@@ -499,6 +477,15 @@ func (r dedicatedClusterResource) Create(ctx context.Context, req resource.Creat
 	}
 	data.TiDBNodeSetting.PublicEndpointSetting = pes
```

Comment:
I think we need to identify publicendpointpoint  is setting or not

Reply:
If publicendpointsetting is not set, the `pes` will be nil